account_invoice_change_currency/models/account_move.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2017-2018 Vauxoo
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    custom_rate = fields.Float(
+        digits=(12, 6),
+        default=1,
+        help="Set new currency rate to apply on the invoice.\n"
+        "This rate will be taken in order to convert amounts between the "
+        "currency on the invoice and last currency",
+    )
+    last_rate = fields.Float(digits=(12, 6), default=1, help="Stores the last rate.")
+    created_invoice_currency_id = fields.Many2one(
+        "res.currency",
+        help="Store the original currency from the company",
+    )
+    last_currency_id = fields.Many2one(
+        "res.currency",
+        help="Store the last currency applied in the invoice",
+    )
+    is_original_currency = fields.Boolean(
+        default=False,
+        store=True,
+        compute="_compute_is_original_currency",
+        help="Check if current currency is the original currency",
+    )
+
+    @api.model
+    def create(self, values):
+        values.setdefault("created_invoice_currency_id", values.get("currency_id"))
+        values.setdefault("last_currency_id", values.get("currency_id"))
+        values.setdefault("last_rate", values.get("custom_rate"))
+        return super(AccountMove, self).create(values)
```

Comment:
Always use new-style super

```suggestion
        return super().create(values)
```

Reply:
Done!